File: src/ServerPulse.Frontend/src/app/modules/authentication/domain/dtos/refreshTokenResponse.ts
import { AccessTokenData } from "../..";

export interface RefreshTokenResponse {
    accessToken: string;
    refreshToken: string;
    refreshTokenExpiryDate: Date;
}

export function mapRefreshTokenResponseToAuthToken(response: RefreshTokenResponse): AccessTokenData {
    return {
        accessToken: response?.accessToken,
        refreshToken: response?.refreshToken,
        refreshTokenExpiryDate: new Date(response?.refreshTokenExpiryDate)
    }
}

File: src/ServerPulse.Backend/AuthenticationApi.Core/Dtos/Endpoints/Auth/RefreshToken/RefreshTokenResponse.cs
namespace AuthenticationApi.Core.Dtos.Endpoints.Auth.RefreshToken
{
    public class RefreshTokenResponse
    {
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime? RefreshTokenExpiryDate { get; set; }
    }
}
